File: app/src/main/java/im/fdx/v2ex/ui/favor/FavorViewPagerAdapter.java
package im.fdx.v2ex.ui.favor;

import android.app.Fragment;
import android.app.FragmentManager;
import android.os.Bundle;
import android.os.Parcelable;
import android.support.v13.app.FragmentPagerAdapter;

import im.fdx.v2ex.ui.main.TopicsFragment;
import im.fdx.v2ex.utils.Keys;

/**
 * Created by fdx on 2017/4/13.
 */
class FavorViewPagerAdapter extends FragmentPagerAdapter {

    FavorViewPagerAdapter(FragmentManager fm) {
        super(fm);
    }


    static final String[] titles = new String[]{"节点收藏", "主题收藏", "特别关注"};

    @Override
    public Fragment getItem(int position) {

        if (position == 0) {
            return new NodeFavorFragment();
        } else {
            TopicsFragment topicFavorFragment = new TopicsFragment();
            Bundle bundle = new Bundle();
            bundle.putInt(Keys.FAVOR_FRAGMENT_TYPE, position);
            topicFavorFragment.setArguments(bundle);
            return topicFavorFragment;
        }
    }

    @Override
    public int getCount() {
        return titles.length;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titles[position];
    }
}

File: app/src/main/java/im/fdx/v2ex/ui/favor/FavorViewPagerAdapter.kt
package im.fdx.v2ex.ui.favor

import android.os.Bundle
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager2.adapter.FragmentStateAdapter
import im.fdx.v2ex.ui.main.TopicsFragment
import im.fdx.v2ex.utils.Keys

/**
 * Created by fdx on 2017/4/13.
 */
internal class FavorViewPagerAdapter(fa: FragmentActivity) : FragmentStateAdapter(fa) {

    companion object {
        val titles = arrayOf( "主题收藏", "节点收藏","特别关注")
    }

    override fun getItemCount(): Int {
        return titles.size
    }

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> NodeFavorFragment()
            else -> {
                TopicsFragment().apply {
                    arguments = bundleOf(Keys.FAVOR_FRAGMENT_TYPE to position)
                }
            }
        }
    }
}
